Automattic/wp-calypso
# Staging Sites: Hide activity log banners during staging site sync

Related to https://github.com/Automattic/dotcom-forge/issues/4177

## Proposed Changes

This PR disables banners in the Activity Log page when a rewind related to sync is in progress.


### Pr main changes.

- Introduced a `sourceSiteTarget` property so we can show from what site we restored ( production or staging ) if we want in the future in the Activity Log ( using the banners ). I also renamed the `siteType` to `targetSite` to be more clear about what this property is.
- Use `isFetchingSyncStatus` so we can avoid displaying the banner on page load ( otherwise it got displayed for a second till the status was fetched )
- Introduced `restoreId` in the sync reducer. I use that to check it the current restore is the same as the latest sync restore. Using that we hide the banners
- Fix a small react warning on `sync-options-panel` component.
- Add a `withSyncStatus` wrapper in order to use that in Activity Log page.
- Fix tests.

## Testing Instructions
1. Apply D125926-code following the instructions there.
2. Create a WoA site and a staging site for that site.
3. Initiate a sync.
4. Navigate to your target's site 'Activity Log' page.
5. Open the network tab and watch the `sync-state` endpoint.
6. Ensure that you don't see any banner when the status becomes 'restoring' (and this remains after that.)
7. Ensure that you don't see any banner when you reload the page.
8. Navigate to the 'Backups' page. 
9. Initiate a backup.
10. Navigate back to the Activity Log page. The `sync-state` endpoint would return 'completed' status, but you ll be able to see the banner.
11. Ensure that you can see the banner after you reload the page.



## Pre-merge Checklist
